### ISSUE TITLE
[BUG] waitForURL timeout causes page to close. test runs until testTimout instead of failing on waitForURL failure

### ISSUE THREAD

**Issue by davidjonesdialexa** [microsoft/playwright#26444]
### System info
- Playwright Version: [v1.37]
- Operating System: [Mac OS 12.6.1, but also seen on Windows 11]
- Browser: [Chrome]
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
// @ts-check
const { defineConfig, devices } = require('@playwright/test');

/**
 * @see https://playwright.dev/docs/test-configuration
 */
module.exports = defineConfig({
  testDir: './tests',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on',
    actionTimeout: 3000,
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'], channel: 'chrome' },
    },
  ],
});

```

**Test file (self-contained)**

```js
test('hotwire', async ({ page }) => {
  await page.goto('https://www.hotwire.com/');
  const hotels = page
    .getByRole('navigation')
    .getByRole('link', { name: 'Hotels' });
  await hotels.click();
  await page.waitForURL(/nike/);
});
```

**Steps**
- Run the test
- check reporter for testTimeout failure rather than waitForURL failure

**Expected**

waitForURL to timeout after 3 seconds

**Actual**

waitForURL closes page and testTimeout of 30 seconds fires

**Comment by davidjonesdialexa** [comment]
I added the following to make sure the actionTimeout did indeed work and the test with this step times out correctly after 3 seconds

```js
const badButton = page.getByRole('button', { name: 'houseses' });
await badButton.click();
```

**Comment by mxschmitt** [comment]
`waitForURL` is doing internally `waitForNavigation`, so the `actionTimeout` does not apply there. Instead the `navigationTimeout` applies in this scenario. When setting the `navigationTimeout` to 3s, it works as expected.

For us waiting is not an action, only clicks, fills, types, etc. are actions. Does this work for you?

**Comment by davidjonesdialexa** [comment]
OH I see... unless I set an explicit navigationTimeout, the testTimeout will take over which is why I'm seeing `Navigation failed because page was closed!`, they occur concurrently. 

In my actual testing suite, I'm doing a trycatch inside a while loop (don't ask, our test data dependency is a real mess and this is how I have to handle it) that waits for a URL, but if waitForURL doesn't fail until the test fails due to test timeout, then the second iteration of the loop never starts. 🤦‍♂️ 

closing as I was misusing the waitFor function. thanks @mxschmitt!
